src/main.py

```diff
@@ -18,28 +20,97 @@ async def on_ready():
     print(client.user.id)
 
 @client.event
-async def on_reaction_add(reaction, user):
-    if reaction.message.channel.id != VERIFY_CHAN or client.user != reaction.message.author:
-        return
+async def on_reaction_add(reaction: discord.Reaction, user: discord.User):
+    # Ignore messages that have been handled previously
     if reaction.message.id in leah.posted:
         return
-    leah.posted.add(reaction.message.id)
-    txt = reaction.message.content.split('\n')
-    other = reaction.message.mentions[0]
-    embed = discord.Embed(title=f"Some amazing art by {str(other)}", type="rich", color=other.color, description=txt[-2])
-    embed.set_image(url=txt[-1])
-    gallery = client.get_channel(GALLERY_CHAN)
-    await gallery.send(embed=embed)
+
+    # Staff reactions to verification channel posts by this client
+    if reaction.message.channel.id == VERIFY_CHAN:
+        # Ignore reactions to posts not by this client in verification channel
+        if client.user != reaction.message.author:
+            return
+
+        # Publish verified posts to gallery channel
+        await publish_art(message=reaction.message)
+
+    # User reactions to posts in self-curated channel
+    elif reaction.message.channel.id in MOD_CHANS:
+        # Ignore reactions from users without any of the required roles
+        if len(SHOWCASE_ROLES) == 0 or len([r for r in user.roles if r.id in SHOWCASE_ROLES]) == 0:
+            return
+
+        # Ignore reactions from users other than the message author
+        if user != reaction.message.author:
+            return
+
+        # Publish self-curated posts
+        await reaction.message.add_reaction(emoji=reaction.emoji)
```

Comment:
This might be fine, but I seem to remember there being oddities with how Discord handles Nitro vs default emojis. Need to double check that this works fine for Nitro stuff.

Reply:
I think there's a member `emoji.is_custom` we can use to ignore ones the bot likely can't use when reposting the message as an embed, and replace them with some generic other emoji. Adding to the same reaction shouldn't cause issues.